src/AppNavigator.tsx

```diff
@@ -201,12 +189,60 @@ const messages = defineMessages({
     id: 'global.actions.dialogs.biometricsChange.message',
     defaultMessage: '!!!Biometrics changed detected ',
   },
+  authorize: {
+    id: 'components.send.biometricauthscreen.authorizeOperation',
+    defaultMessage: '!!!Authorize operation',
+  },
 })
 
-const useAutoLogout = () => {
+const useAutoLogout = (authMethod: AuthMethodState) => {
+  const strings = useStrings()
   const {logout} = useAuth()
+  const {authOsEnabledOnDevice, refetch} = useAuthOsEnabledOnDevice()
+  const osAuthDisabled = !authOsEnabledOnDevice && authMethod.OS
+  const queryClient = useQueryClient()
+
   useBackgroundTimeout({
     onTimeout: logout,
     duration: 120 * 1000,
   })
+
+  React.useEffect(() => {
+    const appStateSubscription = AppState.addEventListener('change', async (appState) => {
+      // when using OS auth and app is active again needs to check if still enabled
+      if (appState === 'active' && authMethod.OS) {
+        queryClient.invalidateQueries(['useAuthAction'])
+        refetch()
+      }
+    })
+    return () => appStateSubscription?.remove()
+  }, [authMethod.OS, queryClient, refetch])
+
+  React.useEffect(() => {
```

Comment:
instead of logging the user out, can we unmount the authenticated root component and display CreatePin, or whatever, then remount the authenticated root component?

Reply:
this is what logout does, it changed the accessible routes.